motion/reference_model/dp_reference_model/src/reference_model.cpp

```diff
@@ -1,102 +1,197 @@
-/*   Written by Jae Hyeong Hwang and Christopher Strøm
-     Modified by Øystein Solbø
-     Copyright (c) 2021 Vortex NTNU.
+/*   Written by Kevin Strandenes and Anders Slåkvik, Student
+     Documentation written by Kevin Strandenes and Anders Slåkvik
+     Copyright (c) 2023 Beluga AUV, Vortex NTNU.
      All rights reserved. */
 
 #include "dp_reference_model/reference_model.h"
+#include "dp_reference_model/eigen_typedefs.h"
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+
+Eigen::Matrix3d skew(Eigen::Vector3d vec) {
+  Eigen::Matrix3d skew_mat = Eigen::Matrix3d::Zero();
+  skew_mat << 0, -vec(2), vec(1), vec(2), 0, -vec(0), -vec(1), vec(0), 0;
+  return skew_mat;
+}
 
-ReferenceModel::ReferenceModel(ros::NodeHandle nh) {
-  // get params
-  double a1, a2, a3;
-  double b1, b2, b3;
-  std::vector<double> beta_temp;
-
-  if (!nh.getParam("dp_rm/a1", a1))
-    a1 = 1;
-  if (!nh.getParam("dp_rm/a2", a2))
-    a2 = -1.990024937655860;
-  if (!nh.getParam("dp_rm/a3", a3))
-    a3 = 0.990049813123053;
-  if (!nh.getParam("dp_rm/b1", b1))
-    b1 = 6.218866798092052e-06;
-  if (!nh.getParam("dp_rm/b2", b2))
-    b2 = 1.243773359618410e-05;
-  if (!nh.getParam("dp_rm/b3", b3))
-    b3 = 6.218866798092052e-06;
-  if (!nh.getParam("/reference_model/dp/beta", beta_temp))
-    beta_temp = {0.025, 0.025, 0.0025};
-
-  a_x = Eigen::Vector3d(a1, a2, a3);
-  b_x = Eigen::Vector3d(b1, b2, b3);
-  beta = Eigen::Vector3d(beta_temp[0], beta_temp[1], beta_temp[2]);
-
-  // initiate local variables
-  prev_control_mode = 0; // OPEN LOOP
-  x_d_prev = Eigen::Vector3d::Zero();
-  x_d_prev_prev = Eigen::Vector3d::Zero();
-  x_ref_prev = Eigen::Vector3d::Zero();
-  x_ref_prev_prev = Eigen::Vector3d::Zero();
+ReferenceModel::ReferenceModel(ros::NodeHandle nh) : m_nh(nh) {
+
+  std::string odometry_topic;
+
+  // Import parameters
+  ReferenceModel::getParameters("dp_rm/zeta_1", zeta_1);
```

Comment:
Same as before, if the ReferenceModel:: here is needed, instead namespace your code to avoid conflicts